### PR TITLE
Fix loading images which aren't divided by 4

### PR DESCRIPTION
**Pull request by Nordsoft91**
Thanks to @IvanSavenko for finding the fix for problems

### CODE REVIEW

**Comment by IvanSavenko** [comment]
Later I with krs found two more cases like this, both are in `ImageLoader::init`
wrong:
`memset((void *)image->bits(), 0, fullSize.y() * fullSize.x());`
correct:
`memset((void *)image->bits(), 0, fullSize.y() * image->bytesPerLine());`

wrong:
`lineStart += margins.y() * image->bytesPerLine() + margins.x();`
correct:

**Comment by krs0** [comment]
Also .pcx single images have some issues I suppose...
Look at last coment in 
VCMI H3 original archives extractor #1118

**Comment by IvanSavenko** [comment]
Yes, pretty much identical bug in `BitmapHandler::loadH3PCX` (BitmapHandler.cpp)
Line 66 right now is;
`QImage image(pcx + it, width, height, QImage::Format_Indexed8);`
should use different constructor:
`QImage image(pcx + it, width, height, width, QImage::Format_Indexed8);`

Line 84 is now:
`QImage image(pcx + it, width, height, QImage::Format_RGB32);`
Should also use different constructor:
`QImage image(pcx + it, width, height, width * 3, QImage::Format_RGB32);`

Somebody ( @krs0  ?) please check whether my assumption is correct.

**Comment by Nordsoft91** [comment]
@IvanSavenko , suspecting last should be
`QImage image(pcx + it, width, height, width * 4, QImage::Format_RGB32);`
right? RGB32 is containing alpha

**Comment by IvanSavenko** [comment]
Oh, yeah. But actually not exactly. Both versions are wrong.
Should be:
`QImage image(pcx + it, width, height, width * 3, QImage::Format_RGB888);`
H3 PCX data is store in RGB format, 24 bits per pixel. Qt provides such format as Format_RGB888.

**Comment by krs0** [comment]
and please add also for both ifs `return image.copy();` as per Ivan

File: mapeditor/BitmapHandler.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ namespace BitmapHandler
 		}
 		else
 		{
-			QImage image(pcx + it, width, height, QImage::Format_RGB32);
+			QImage image(pcx + it, width, height, width * 3, QImage::Format_RGB888);
```
Comment (nullkiller) [inline mapeditor/BitmapHandler.cpp:84]: 3 is 3 bytes?

**Review by nullkiller — APPROVED** [review]
Looks good to merge

File: mapeditor/BitmapHandler.cpp (hunk 1 above)
Comment (Nordsoft91) [reply mapeditor/BitmapHandler.cpp:84]: Yes, that's byter per line as I understand

**Pull request merged**